File: easysql_web/src/api/execute.ts
import { apiClient } from './client';
import type { ExecuteRequest, ExecuteResponse, SqlCheckResult } from '@/types';

export const executeApi = {
  executeSql: async (data: ExecuteRequest): Promise<ExecuteResponse> => {
    const response = await apiClient.post<ExecuteResponse>('/execute', data);
    return response.data;
  },

  checkSql: async (data: ExecuteRequest): Promise<SqlCheckResult> => {
    const response = await apiClient.post<SqlCheckResult>('/execute/check', data);
    return response.data;
  },
};

File: easysql_api/routers/execute.py
"""
SQL Execution Router.

Standalone endpoint for executing SQL queries, decoupled from the agent workflow.
"""

from __future__ import annotations

from typing import Annotated

from fastapi import APIRouter, Depends

from easysql_api.deps import get_execute_service_dep
from easysql_api.models.execute import (
    ExecuteRequest,
    ExecuteResponse,
    SqlCheckResult,
)
from easysql_api.services.execute_service import ExecuteService

router = APIRouter()


@router.post("/execute", response_model=ExecuteResponse)
async def execute_sql(
    request: ExecuteRequest,
    service: Annotated[ExecuteService, Depends(get_execute_service_dep)],
) -> ExecuteResponse:
    return service.execute(request)


@router.post("/execute/check", response_model=SqlCheckResult)
async def check_sql(
    request: ExecuteRequest,
    service: Annotated[ExecuteService, Depends(get_execute_service_dep)],
) -> SqlCheckResult:
    return service.check_sql(request.sql)
